extension/pinch-zoom.js

```diff
@@ -185,14 +229,15 @@ function updateTransform(scaleModeOverride, shouldDisableControl) {
 
 	if (shouldDisableControl) {
 		disableControl();
-		clearTimeout(overflowTimeoutHandle);
-		overflowTimeoutHandle = setTimeout(function(){
+		cancelAnimationFrame(overflowRequestHandle); //cancels previous request before implementing new one
+		//uses requestAnimationFrame to prevent glitches with setTimeout
+		overflowRequestHandle = requestAnimationFrame((e) => {
 			restoreControl();
-		}, 400);
```

Comment:
Unfortunately removing this timeout period causes performance issues for me (on 2016 mbp). The problem is that if scrollbars are visible when the page scale changes then Firefox does a complete repaint of the page which is expensive. To hack around this I disable scrollbars while zooming (unfortunately we have to use a magic number with a trackpad because we can't actually determine when the user has stopped zooming – this is not the case for touchscreens however).

By re-enabling scrollbars every frame we're repainting all the time which is the cause of the perf problems :(

One fix would be to have a separate path for touchscreen gestures, where it executes `restoreControl()` on touch-up

A better fix would be to figure out a way to prevent the re-paint when scale changes

Reply:
Sorry, I did not notice any performance impact. (I am on Win10 with 2015 HP Spectre x360)
I have reverted the changes, except when using the touchscreen.